### ISSUE TITLE
Multi renderer backend support for Brux

### ISSUE THREAD

**Issue by tulpenkiste** [KelvinShadewing/brux-gdk#69]
This issue simply exists so I can remember to work on this for brux in the near future.

Renderers on initial completion:
- SDL2 + SDL2_gfx
- SDL3 + SDL GPU (allows for shaders) (the base SDL3 backend needs to be backwards compatible with SDL2, SDL GPU cannot as thats a new feature to SDL3)

I will probably split this into 2 PRs and this will be started after #66 is completed and merged into unstable.

**Comment by hexaheximal** [comment]
I wrote the audio portability code with that goal in mind but never got around to doing the graphics part.

Here's the problem though: SuperTux Advance utilizes A LOT of complex SDL2 functionality for graphics. I don't know how you would abstract that away without a significant amount of effort...

- hexaheximal

**Comment by tulpenkiste** [comment]
Due to the recent [merging of SimpleSquirrel support](https://github.com/KelvinShadewing/brux-gdk/pull/71), I'll be taking a crack at this later
